### DISCUSSION TITLE
Problem with the removal of combineLatest in version 8

### DISCUSSION THREAD

**Discussion by scottburch** [ReactiveX/rxjs#7495] [Q&A]
Hello,

I have been using RxJS for pretty much all of my code for years now.  I'm not new, but not a total expert either.

I read that combineLatest is scheduled to be removed in version 8.  However, I use it for something that I can't find anything else to do the same job.  I use it to pass in a value into xMap and return multiple results.  Basically, a passthrough.  I put some code below to demonstrate how I use it

The alternative to this is adding a .pipe() at the end of the newTransaction, and forming the tuple.  However, I find that to be a bit messier.  If someone has a better way to do this, let me know.  But, if combineLatest is removed, I would need to find a way to do this without it.  

Thanks

```                
                 switchMap(keys => combineLatest([
                    of(keys), newTransaction({
                        msgs: [{path: 'my/transaction', data: {foo: 10}}]
                    })
                ])),
                switchMap(([keys, tx]) => signTx(tx, keys)),
```

**Comment by AronHoxha** [comment]
Hi @scottburch !
For your particular example I think you can utilize `combineLatestWith`, as follows:
```ts
                 // observable that resulted to keys ...
                 combineLatestWith(newTransaction({
                    msgs: [{path: 'my/transaction', data: {foo: 10}}]
                })
                switchMap(([keys, tx]) => signTx(tx, keys)),
```
Assuming that `newTransaction()` function is not dependent on the keys, nor is it supposed to be lazily evaluated

In addition AFAIK, combineLatest is not being totally removed in v8 (or I would also be very sad :sweat_smile: ), just some specific implementation of it are, for instance rest parameters implementation:
```ts
combineLatest(
  of(1),
  of(2),
  of(3),
  // ...
)
```

Hope this helps.

**Comment by scottburch** [comment]
@AronHoxha Thanks for the reply.

Yea, I thought that combineLatestWith would work, but it does not do the same thing for some reason.  I can't remember why.   But, I think I ran into the same thing after I posted, that the array argument implementation was not being removed, just the ...rest implementation.  So, all should be good.

Thanks again.  I really appreciate it.  I can try the combineLatestWith again if I need to, or try to figure out why it did not work the same.
